File: SoccerTeamManagement/ClientApp/src/app/interfaces/address.ts
import { IEntity } from './entity';
import { ICountry } from './lookups/country';
import { IState } from './lookups/state';

export interface IAddress extends IEntity {
  addressLine1: string;
  addressLine2: string;
  city: string;
  zipCode: string;

  country: ICountry;
  countryId: number;
  countryText: string;
  countryAlpha2Code: number;
  state: IState;
  stateId: number;
  stateText: string;
  stateAlpha2Code: number;
}

File: 1.Core/Models/Address.cs
using Core.Abstractions;
using Core.Models.Lookups;

namespace Core.Models
{
    public class Address : EntityBase
    {
        #region Properties

        public string AddressLine1 { get; set; }

        public string AddressLine2 { get; set; }

        public string City { get; set; }

        public string ZipCode { get; set; }

        #endregion Properties

        #region Relationships

        public int? CountryId { get; set; }

        public Country Country { get; set; }

        public int? StateId { get; set; }

        public State State { get; set; }

        #endregion Relationships
    }
}